src/main.rs

```diff
@@ -324,73 +161,64 @@ async fn main() {
         return;
     }
 
-    println!("{}", "Fetching and validating the proof itself".on_blue());
-
-    let proof_response = requests::fetch_proof_from_storage(batch_number, &network).await;
-
-    if let Err(_err) = proof_response {
-        println!("{}", _err);
-        return;
-    }
-    let proof_path = proof_response.unwrap();
-
-    // First, we verify that the proof itself is valid.
-    let valid_public_inputs = verify_scheduler_proof(&proof_path, batch_number);
-    if valid_public_inputs.is_err() {
-        println!("Proof is {}", "INVALID".red());
-        return;
-    } else {
-        println!("Proof is {}", "VALID".green());
-    }
-    println!("\n");
+    check_should_download_verification_key(opt.update_verification_key).await;
 
+    println!("{}", "Fetching and validating the proof itself".on_blue());
     if l1_rpc.is_none() {
         println!(
             "{}",
             "Skipping building batch information from Ethereum as no RPC url was provided."
                 .yellow()
         );
     } else {
-        // Then we verify the public inputs (that contain the circuit code, prev and next root hash etc)
-        // To make sure that the proof is matching a correct computation.
+        let contract = ContractConfig::new(l1_rpc.clone().unwrap(), network.clone());
 
-        let public_inputs = compute_public_inputs(network, batch_number, l1_rpc)
-            .await
-            .unwrap();
+        let L1BatchAndProofData {
+            aux_output,
+            scheduler_proof,
+            batch_l1_data,
+            verifier_params,
+            block_number,
+        } = requests::fetch_l1_data(batch_number, &network, &l1_rpc.clone().unwrap()).await;
 
-        let valid_public_inputs = valid_public_inputs.unwrap();
+        let vk_hash = contract.get_verification_key_hash(block_number).await;
 
-        println!(
-            "{} ",
-            "Comparing public input from Ethereum with input for boojum".on_blue()
-        );
-        println!(
-            "Recomputed public input from current prover using L1 data is {}",
-            format!("{:?}", public_inputs).bright_blue()
-        );
-        println!(
-            "Boojum proof's public input is {}",
-            format!("{:?}", valid_public_inputs).green()
-        );
+        let snark_vk_scheduler_key_file = "src/keys/scheduler_key.json";
 
-        if public_inputs == valid_public_inputs {
-            println!("Boojum's proof is {}", "VALID".green());
-        } else {
-            println!("Boojum's proof is {}", "INVALID".red());
-        }
-        return;
-    }
+        let mut batch_proof = L1BatchProofForL1 {
+            aggregation_result_coords: aux_output.prepare_aggregation_result_coords(),
+            scheduler_proof,
+        };
+
+        let inputs = generate_inputs(batch_l1_data, verifier_params);
+
+        batch_proof.scheduler_proof.inputs = inputs;
```

Comment:
didn't the scheduler_proof already contain inputs ? (then we could simply just compare them here )